node/src/blockchain/test_utils.rs

```diff
@@ -56,9 +63,139 @@ pub fn make_meaningless_seed() -> Seed {
+        self.transaction_index_opt = Some(index);
+        self
+    }
+
+    pub fn block_hash(mut self, hash: Hash) -> ReceiptResponseBuilder {
+        self.block_hash_opt = Some(hash);
+        self
+    }
+
+    pub fn block_number(mut self, number: U64) -> ReceiptResponseBuilder {
+        self.block_number_opt = Some(number);
+        self
+    }
+
+    pub fn cumulative_gas_used(mut self, number: U256) -> ReceiptResponseBuilder {
+        self.cumulative_gas_used_opt = Some(number);
+        self
+    }
+
+    pub fn gas_used(mut self, number: U256) -> ReceiptResponseBuilder {
+        self.gas_used_opt = Some(number);
+        self
+    }
+
+    pub fn contract_address(mut self, hash: H160) -> ReceiptResponseBuilder {
+        self.contract_address_opt = Some(hash);
+        self
+    }
+
+    pub fn logs(mut self, logs: Vec<Log>) -> ReceiptResponseBuilder {
+        self.logs_opt = Some(logs);
+        self
+    }
+
+    pub fn status(mut self, number: U64) -> ReceiptResponseBuilder {
+        self.status_opt = Some(number);
+        self
+    }
+
+    pub fn root(mut self, hash: Hash) -> ReceiptResponseBuilder {
+        self.root_opt = Some(hash);
+        self
+    }
+
+    pub fn logs_bloom(mut self, bloom: H2048) -> ReceiptResponseBuilder {
+        self.logs_bloom_opt = Some(bloom);
+        self
+    }
+
+    pub fn build(self) -> String {
+        let mut transaction_receipt = TransactionReceipt::default();
+
+        if let Some(transaction_hash) = self.transaction_hash_opt {
+            transaction_receipt.transaction_hash = transaction_hash;
+        }
+
+        if let Some(index) = self.transaction_index_opt {
+            transaction_receipt.transaction_index = index;
+        }
+
+        if let Some(cumulative_gas_used) = self.cumulative_gas_used_opt {
+            transaction_receipt.cumulative_gas_used = cumulative_gas_used;
+        }
+
+        if let Some(logs) = self.logs_opt {
+            transaction_receipt.logs = logs;
+        }
+
+        if let Some(bloom) = self.logs_bloom_opt {
+            transaction_receipt.logs_bloom = bloom;
+        }
+
+        transaction_receipt.block_hash = self.block_hash_opt;
+        transaction_receipt.block_number = self.block_number_opt;
+        transaction_receipt.gas_used = self.gas_used_opt;
+        transaction_receipt.contract_address = self.contract_address_opt;
+        transaction_receipt.status = self.status_opt;
+        transaction_receipt.root = self.root_opt;
+
+        let rpc_response = RpcResponse {
+            json_rpc: "2.0".to_string(),
+            id: 0,
+            result: transaction_receipt,
+        };
+        serde_json::to_string(&rpc_response).unwrap()
+    }
+}
+
 #[derive(Default)]
 pub struct BlockchainInterfaceMock {
-    retrieve_transactions_parameters: Arc<Mutex<Vec<(BlockNumber, BlockNumber, Wallet)>>>,
+    retrieve_transactions_parameters: Arc<Mutex<Vec<(BlockNumber, u64, Address)>>>,
```

Comment:
Again, this is more addressed to Utkarsh, just looking at the types, I don't feel like it was a change to the better. This kind of types joggling is not really my cup of tea.